DEV-672 - Concept page for delegated repositories (#2205)
gruntwork-io/docs: docs/2.0/docs/accountfactory/concepts/delegated-repositories.md

# Delegated Repositoires

:::note

> Delegated Repositories are only available to DevOps Foundations Enterprise customers.

Reviewer: Our wording here is really tricky. 

You can see [here](https://github.com/gruntwork-io/docs/blob/main/docs/2.0/docs/pipelines/guides/setup-delegated-repo.mdx) that I tried to call this "Account Factory vended delegated repos" but maybe we need a new term for what that that other thing is. Users can have some amount of infrastructure delegation without being enterprise customers, they just don't get Account Factory vending them as part of account creation.

Whatever we call the other thing, it's an important thing for users to know about, as many want to know how to handle things like application deployments that require AWS access to push artifacts, etc, even if they aren't managing IaC.

:::

As enterprises scale their usage of IaC across more teams there often arises a need to enforce least-privilege access control to both IaC source code and the CI/CD of that code. Gruntwork's recommendation is to leverage multiple source code repositories to create hard-boundaries that enforce least-privilege access control. The pattern we recommend consists of three parts:

1. A single `infrastructure-live-root` repository, governed by the core platform team. This repository includes IaC for security and shared infrastructure including core-account creation, central logging/auditing and networking controls (e.g. GuardDuty, SecurityHub, Macie, Transit Gateway etc.).

> 1. Many _delegated_ repositories, often named using the pattern `infrastructure-live-$TEAM_NAME`, where individual teams can manage their own IaC. These repositories are created by the core platform team using the Account Factory and, via `infrastructure-live-access-control` are granted permissions to manage their own infrastructure.

Reviewer 2: ```suggestion
1. Many _delegated_ repositories, often named using the pattern `infrastructure-live-$TEAM_NAME`, where individual teams can manage their own IaC. These repositories are created using the Account Factory and are granted permissions to manage their own infrastructure via `infrastructure-live-access-control`.
```

> 1. Many _delegated_ repositories, often named using the pattern `infrastructure-live-$TEAM_NAME`, where individual teams can manage their own IaC. These repositories are created by the core platform team using the Account Factory and, via `infrastructure-live-access-control` are granted permissions to manage their own infrastructure.

Reviewer: ```suggestion
1. Many _delegated_ repositories, often named using the pattern `infrastructure-live-$TEAM_NAME`, where individual teams can manage their own IaC. These repositories are created by the core platform team using the Account Factory and, via `infrastructure-live-access-control` are granted limited permissions to manage their own infrastructure.
```

> git

Reviewer 3: ```suggestion

```

> git

Reviewer 2: ```suggestion

```
Typo?

> Using this pattern central platform teams have the ability to:

Reviewer 2: ```suggestion
Using this pattern core platform teams have the ability to:
```
For consistency

* Centrally manage core compliance and security infrastructure

> * Centrally manage which teams have permission to deploy what kind of infrastructure, allowing central teams to enforce compliance with approved architecture-board approved plans

Reviewer 3: ```suggestion
* Centrally manage which teams have permission to deploy what kind of infrastructure, allowing their team to enforce compliance with architecture-board approved plans
```

Reviewer: I would recommend we just explicitly say "allowing the central platform team to enforce". Using different terms for the team doing the regulation gets hard to follow.

> * Centrally manage which teams have permission to deploy what kind of infrastructure, allowing central teams to enforce compliance with approved architecture-board approved plans

Reviewer 2: ```suggestion
* Centrally manage which teams have permission to deploy different types of infrastructure, allowing core teams to enforce compliance with approved architecture-board approved plans
```

* Centrally manage tags for shared and per-team infrastructure

> * Enforce least-privilege access to IaC, restricting teams to only have access only to IaC and deployment capability / role assumptions for resources that they are responsible for.

Reviewer 3: ```suggestion
* Enforce least-privilege access to IaC, restricting teams to have access only to IaC and deployment capability / role assumptions for resources that they are responsible for.
```

## Delegated Repository Creation

> Delegated repositories are optionally created by [Account Factory](http://localhost:3000/2.0/docs/accountfactory/concepts/) during account creation.  A delegated account vend follows the following (automated) workflow:

Reviewer 3: ```suggestion
Delegated repositories are optionally created by [Account Factory](/2.0/docs/accountfactory/concepts/) during account creation.  A delegated account vend follows the following (automated) workflow:
```

> Delegated repositories are optionally created by [Account Factory](http://localhost:3000/2.0/docs/accountfactory/concepts/) during account creation.  A delegated account vend follows the following (automated) workflow:

Reviewer 2: ```suggestion
Delegated repositories are optionally created by [Account Factory](/2.0/docs/accountfactory/concepts) during account creation.  A delegated account vend follows the following (automated) workflow:
```
